src/webapi/rendering_context.rs

```diff
@@ -27,7 +183,771 @@ impl RenderingContext for CanvasRenderingContext2d {
     }
 }
 
+impl CanvasGradient {
+
+    /// Adds a new stop, defined by an offset and a color, to the gradient. If the offset is 
+    /// not between 0 and 1, an INDEX_SIZE_ERR is raised, if the color can't be parsed as a 
+    /// CSS <color>, a SYNTAX_ERR is raised.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasGradient/addColorStop)
+    // https://html.spec.whatwg.org/#2dcontext:dom-canvasgradient-addcolorstop
+    pub fn add_color_stop(&self, offset: f64, color: &str) -> Result<(), SyntaxError> {
+        assert!(offset > 0 as f64 && offset < 1 as f64);
+        js_try! ( @(no_return)
+            @{&self.0}.addColorStop(@{offset}, @{color});
+        ).unwrap()
+    }
+}
+
+impl ImageData {
+
+    /// Returns a Uint8ClampedArray representing a one-dimensional array containing the data in the RGBA order, 
+    /// with integer values between 0 and 255 (included).
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/ImageData/data)
+    pub fn get_data(&self) -> Vec<u8> {
```

Comment:
This should return `Uint8ClampedArray`, not `Vec<u8>`.

Right now we don't have any way of expressing `Uint8ClampedArray`. We have `TypedArray< u8 >`, but that's a non-clapmed variant. So until we have that you can temporairly comment this function out, leave a TODO and we can fix it later after we merge this PR.